services/application/src/actions/user/send-login-link.js

```diff
@@ -61,7 +62,11 @@ module.exports = async ({
 
   if (supportEmail) addressValues.push(supportEmail);
 
-  const { token } = await createLoginToken({ applicationId, email: user.email, data: { source } });
+  const { token } = await createLoginToken({
+    applicationId,
+    email: user.email,
+    data: { source, additionalEventData },
```

Comment:
Spread this. Since `source` is more specific, add it after the spread of your data so it cannot be overwritten.